### PR TITLE
Feat/download btn

### PR DESCRIPTION
**Pull request by utku-guclu**
- Button downloads PDF related to exam.
- Refactored with redux
~- Second try getting "undefined" error, _I don't know what is the reason behins it_.~
- _Loading status also not working._
- _I couldn't change the status of styled components_ 
[ticket](https://trello.com/c/J9GTMqyy/1020-exam-functionality-download-button-download-exam)

### CODE REVIEW

**Comment by johnnetr2** [comment]
@utku-guclu After clicking on download button, it downloads but the App is frozen and doesnt work. Fix this and I think we are good afterward.

**Comment by utku-guclu** [comment]
> 

@johnnetr2 I guess it is working correctly now.

File: playwright-report/index.html
Comment (lucasgojeda) [inline playwright-report/index.html:1]: The playwright reports should be into the .gitignore and we shouldn't send them to the repository

File: src/assets/Static/allProvPassDates.json
Comment (lucasgojeda) [inline src/assets/Static/allProvPassDates.json:1]: Nice approach here!

File: src/components/organism/CoursesOrg/CoursePages/TestInformation/TestInformation.js
Comment (lucasgojeda) [inline src/components/organism/CoursesOrg/CoursePages/TestInformation/TestInformation.js:1]: Good implementation of a custom hook here!

File: src/config/development-config.js
Comment (lucasgojeda) [inline src/config/development-config.js:1]: Please change back this, we shouldn't touch this, I think this is related to the deployment process Rashdan is the one on charge of this

File: src/redux/slices/index.js
Comment (lucasgojeda) [inline src/redux/slices/index.js:1]: Nice that you saw the barrel file we should use them always that we can!

File: src/redux/slices/pdfDownload/pdfDownloadSlice.js
Comment (lucasgojeda) [inline src/redux/slices/pdfDownload/pdfDownloadSlice.js:1]: This slice should be in the "/slices/utils" folder and we shouldn't import things to work with slices, if we need data we send it through props when we call the reducer in the custom hook and use them from the action.payload into the slices/reducers

Comment (lucasgojeda) [reply src/redux/slices/pdfDownload/pdfDownloadSlice.js:1]: The reducers should be pure functions

https://redux.js.org/understanding/thinking-in-redux/three-principles#:~:text=Reducers%20are%20just%20pure%20functions,of%20mutating%20the%20previous%20state.

https://stackoverflow.com/questions/44767160/why-are-pure-reducers-so-important-in-redux

File: src/redux/slices/pdfDownload/utils/pdfDownloadSliceUtils.js
Comment (lucasgojeda) [inline src/redux/slices/pdfDownload/utils/pdfDownloadSliceUtils.js:1]: This file should be in other place, maybe into a custom hook

**Comment by lucasgojeda** [comment]
Everything looks quite better, great job Utku, just a few details, let me know if you understand what are about if now, we could have quick call and I can explain you better in 10 minutes

**Comment by lucasgojeda** [comment]
I changed the gitignore and fix the merge conflicts

**Comment by edgarpalic** [comment]
Really nice work. 

I just have some user experience feedback. When I download multiple exams at once and then look at the files they all have the same name. It would be better if the files shared the name of the exam. Varen2022april.zip. Kind of like that.

fileNames

**Comment by utku-guclu** [comment]
> Everything looks quite better, great job Utku, just a few details, let me know if you understand what are about if now, we could have quick call and I can explain you better in 10 minutes

Thank you, Lucas! I appreciate your feedback. I believe a 10-minute call would be beneficial to ensure I fully understand the details you mentioned. Please let me know a convenient time for the call.

**Comment by utku-guclu** [comment]
> Really nice work.
> 
> I just have some user experience feedback. When I download multiple exams at once and then look at the files they all have the same name. It would be better if the files shared the name of the exam. Varen2022april.zip. Kind of like that.

Thank you, Edgar! I appreciate your feedback on the user experience. It is a good idea! I will work on implementing that improvement.

File: .env
Comment (lucasgojeda) [inline .env:1]: Please don't change the api base

File: test-results/prognos-prognos-Test-prognos-number-equality-test-chromium/trace.zip
Comment (lucasgojeda) [inline test-results/prognos-prognos-Test-prognos-number-equality-test-chromium/trace.zip:1]: Please remove this file from the PR

**Comment by lucasgojeda** [comment]
Only two comments this time, I like more the new approach, I can see that you start to learn to work with custom hooks and redux! More redux tasks coming in the future!

**Review by edgarpalic — APPROVED** [review]

**Comment by johnnetr2** [comment]
Very good @utku-guclu So work on those comments/feedbacks above and I also sent you a video on Slack where I gave you one additional feedback + an approach that may solve the CORS errors you have for older exams.

**Review by johnnetr2 — APPROVED** [review]
Tested the feature, works great! Very well done @utku-guclu .

@lucasgojeda Could you check, if there is no further feedback from you then you can approve and merge this today as well.

**Review by lucasgojeda — APPROVED** [review]

**Comment by lucasgojeda** [comment]
Looks great now, you just have to solved the conflicts and then we can merge it!

**Comment by lucasgojeda** [comment]
@utku-guclu Please fix the conflicts and then we can merge it!

**Comment by utku-guclu** [comment]
> @utku-guclu Please fix the conflicts and then we can merge it!

I fixed it.

**Pull request merged**